### DISCUSSION TITLE
Autosuggest

### DISCUSSION THREAD

**Discussion by boxadesign** [ONSdigital/design-system#894] [Feedback]
Use this place to discuss the [Autosuggest](https://ons-design-system.netlify.com/components/autosuggest) in the [ONS Design System](https://ons-design-system.netlify.com)

**Comment by Paul-Joel** [comment]
There is a jinja issue with this component.
The code uses 'params.uprn.value' in the hidden uprn input field, however this doesn't exist on page load, so the jinja fails.
A value needs defining to use as a default for the initial load, which is then updated with the actual value on submission

**Reply by jrbarnes9** [reply]
Fixed in #1605

**Comment by Paul-Joel** [comment]
Currently it appears that you can only use an 'external' json file for the autosuggest 'list'. 
Could this be enhanced to allow it to alternatively use a json 'blob' instead? This would allow for the option to dynamically build the autosuggest list from dynamic query data, ie if the UI running the DS received a dynamic set of data about different surveys, the autosuggest should be able to receive a json blob containing the different surveys in the dynamic data, rather than being forced to have a predetermined json file.
Whilst it could be possible to generate a dynamic json output at a fixed url, it would be easier to be able to pass the json directly to the autosuggest component

**Comment by mitz-lad** [comment]
Has there been any accessibility issues discovered when using the autosuggest? Wondering what the experience is like for any text to speech, JAWS and Dragon users.

**Comment by jrbarnes9** [comment]
Autosuggest incorrectly using `<header>` element

Using the HTML `<header>` element for the pop-up search listbox heading 'Search results' is not semantically correct as it is not a document or section heading. It produces invalid HTML when nested inside the page `header`

**Reply by jrbarnes9** [reply]
Raised #2620

**Comment by jrbarnes9** [comment]
Autosuggest missing `Aria-atomic="true"` on status container

Reported by Site Improve:
> "The page contains an error container made with the WAI-ARIA role="alert" or aria-live="assertive" attributes, but it does not contain a WAI-ARIA attribute that will make sure all assistive technologies can read the error message after more than one invalid submission.

> To make sure that all assistive technologies will read the error message after more than one invalid submission, you need to add the WAI-ARIA attribute aria-atomic="true" to the error container tag."

https://www.w3.org/TR/WCAG20-TECHS/ARIA19.html

**Reply by jrbarnes9** [reply]
Raised #2622

**Comment by Al-Davies** [comment]
**UR findings on auto-suggest**
tested as part of end-to-end Census testing Feb-Sept 2020

- mixed use of this for Census 2021, sone users would type a few characters and then pick from the drop-down that appeared, some just carried on typing the full name
- some confusion seen if users type the full name of a place and then that doens't appear in some form of drop-down list - they worry that the word they have typed isn't valid - this could be a particular issues for job titles
- if users are able to type in 2 answers ie. in Census they could type in 2 Countries for passports, we need to be clear that they need to add a , between entries, or provide them with multiple boxes to type multiple answers
- users find auto-suggest helpful if their spelling is poor

**Comment by cconway-ext-ons** [comment]
Only the first 30 characters in the list seem to be searched.
This can be seen in the countries example if you search for **South Georgia and the South Sandwich Islands**
which appears in search for **South S**, but not **South Sa** 

![Screenshot 2024-07-24 at 08 59 19](https://github.com/user-attachments/assets/0b96e7f2-759d-479c-a561-9bec2118240b)
![Screenshot 2024-07-24 at 08 59 30](https://github.com/user-attachments/assets/be1386a8-b9c4-4559-8f9f-696d1cbcc75e)

**Reply by MagdalenaLarge** [reply]
Hi @cconway-ext-ons,
Thank you for raising this issue.
Team is currently looking at it. We need to identify the root cause and the best solution. 
I will come back to you once we agree the next action. 
Thank you for your patience.
